utils/scripts/load-binary.sh

```diff
@@ -205,14 +205,42 @@ elif [ "$TARGET" = "ruby" ]; then
 
 elif [ "$TARGET" = "php" ]; then
     rm -rf *.tar.gz
+    mkdir -p temp
     if [ $VERSION = 'dev' ]; then
-        ../utils/scripts/docker_base_image.sh ghcr.io/datadog/dd-trace-php/dd-library-php:latest_snapshot ./temp
+        LIBRARY_TARGET_BRANCH="${LIBRARY_TARGET_BRANCH:-master}"
+        COMMIT_ID=$(curl -sS --fail "https://api.github.com/repos/DataDog/dd-trace-php/branches/$LIBRARY_TARGET_BRANCH" | jq -r .commit.sha)
+        VERSION=$(curl -sS --fail "https://raw.githubusercontent.com/DataDog/dd-trace-php/refs/heads/$LIBRARY_TARGET_BRANCH/VERSION")
+
+        # if we have e.g. a beta suffix, just strip it
+        if [[ $VERSION == *-* ]]; then
+            VERSION=${VERSION%-*}
+        else
+            # otherwise increment minor VERSION
+            parts=($(echo -n "$VERSION" | tr '.' '\n'))
+            parts[1]=$((parts[1]+1))
+            parts[2]=0
+            VERSION=$(export IFS=.; (echo "${parts[*]}"))
+        fi
+        VERSION_HASH="$VERSION+$COMMIT_ID"
+        VERSION_HASH_ENCODED="$VERSION%2B$COMMIT_ID"
+
+        echo "Using git commit $COMMIT_ID"
+        echo "Downloading version $VERSION"
+
+        URL="https://s3.us-east-1.amazonaws.com/dd-trace-php-builds/${VERSION_HASH_ENCODED}/datadog-setup.php"
+        echo "Downloading datadog-setup.php from: $URL"
+        curl --fail --location --silent --show-error --output ./temp/datadog-setup.php "$URL"
```

Comment:
This is fundamentally correct, but you have to take into account that there might be versions on top of master, which haven't run CI yet to the point (or failed...).

You'll have to do a for loop through the commit history (starting with the $COMMIT_ID you fetched), to try its parent commit (for merge commits use the first parent) or even further ancestor commits.

Reply:
Ok. Anyways, do you reckon we should make this side more complex? Not sure if we can find an easier way to upload artifacts that from here it's just picking the `latest`